types/logger.d.ts

```diff
@@ -21,6 +21,8 @@ export type ChildLoggerOptions = pino.ChildLoggerOptions
 
 export type FastifyBaseLogger = pino.BaseLogger & {
   child(bindings: Bindings, options?: ChildLoggerOptions): FastifyBaseLogger
+  bindings(): Bindings
```

Comment:
It is not necessary for base logger.
I would just update the `default assignment`.

Reply:
Done

Reply:
My suggestion is not clear enough.
I means you should create a new type for `DefaultFastifyLogger`.
Then, you should update the `default assignment`.

Reply:
1. Create new type: `DefaultFastifyLogger`
2. Update `FastifyBaseLogger` type to `export type FastifyBaseLogger = pino.BaseLogger & DefaultFastifyLogger`
Is that what you mean?

Reply:
`BaseLogger` should not be a instance of `DefaultFastifyLogger`.

Replace the current `FastifyLoggerInstance` to `DefaultFastifyLogger` except the line below.
https://github.com/fastify/fastify/blob/77b75820be399d8d28bbda676c130ee2b90c3f52/types/loggDefaultFastifyLoggerzer.d.ts#L26-L31

Reply:
OK. So we need replace `FastifyBaseLogger` with `DefaultFastifyLogger`? Looks like this is a rename operation?

Reply:
`export type FastifyLoggerInstance = DefaultFastifyLogger`

`FastifyBaseLogger` remain unchanged.
`DefaultFastifyLogger` is the default logger instance used by `fastify` which can be extends the `pino.Logger`.

Reply:
If the works are too many, only change the `FastifyLoggerInstance` types is acceptable.

```ts
export interface FastifyLoggerInstance extends FastifyBaseLogger, pino.Logger {

}
```

Reply:
It looks like the code should look something like the following.

```ts
export type DefaultFastifyLogger<Options = pino.LoggerOptions> = FastifyBaseLogger & pino.Logger<Options>
export type FastifyLoggerInstance = DefaultFastifyLogger;
```

But in doing so, we will get a lot of type errors. See:

<img width="1413" alt="image" src="https://user-images.githubusercontent.com/8198408/196098965-9f368b25-ae77-44f6-89a0-e514611ca05b.png">

Reply:
You should use `interface <> extends <>` to eliminate the types compatibility problem.

Reply:
1.
```ts
export type FastifyBaseLogger = pino.BaseLogger

export interface DefaultFastifyLogger extends FastifyBaseLogger, Omit<pino.Logger, "child"> {
  setBindings(bindings: Bindings): void
  child(bindings: Bindings, options?: ChildLoggerOptions): FastifyBaseLogger
}

// TODO delete FastifyBaseLogger in the next major release. It seems that it is enough to have only FastifyBaseLogger.
/**
 * @deprecated Use FastifyBaseLogger instead
 */
export type FastifyLoggerInstance = DefaultFastifyLogger
```

2. replace `FastifyBaseLogger` with `DefaultFastifyLogger`. Because following:
    <img width="1054" alt="imgae" src="https://user-images.githubusercontent.com/8198408/196101551-b4cb941c-6d5d-4041-a0bd-783e53b926b7.png">

Is that right?

Reply:
`FastifyBaseLogger` required to have `.child` method.

You should not replace `FastifyBaseLogger` with `DefaultFastifyLogger`.
`FastifyBaseLogger` must remain unchanged.

Reply:
🤔 So the code looks like is following:

```ts
export type FastifyBaseLogger = pino.BaseLogger & {
  child(bindings: Bindings, options?: ChildLoggerOptions): FastifyBaseLogger
}

export interface DefaultFastifyLogger extends FastifyBaseLogger, Omit<pino.Logger, "child"> {
  setBindings(bindings: Bindings): void
}

// TODO delete FastifyBaseLogger in the next major release. It seems that it is enough to have only FastifyBaseLogger.
/**
 * @deprecated Use FastifyBaseLogger instead
 */
export type FastifyLoggerInstance = DefaultFastifyLogger
```
----

> replace FastifyBaseLogger with DefaultFastifyLogger. Because following:

I don't mean renaming and mean the following:

<img width="792" alt="image" src="https://user-images.githubusercontent.com/8198408/196103808-4987b293-67e3-4f62-a77f-4471a1717a47.png">

Otherwise, we will get a type error:

<img width="1054" alt="imgae" src="https://user-images.githubusercontent.com/8198408/196101551-b4cb941c-6d5d-4041-a0bd-783e53b926b7.png">

Reply:
You can add an extra argument for it. Like the `Request`
https://github.com/fastify/fastify/blob/b5e4157e33d2c96bd931c1adc6306afab8727c82/types/request.d.ts#L33

Reply:
Do you mean is add the `logger` key in the `reply` object?

<img width="485" alt="image" src="https://user-images.githubusercontent.com/8198408/196105779-79dd7bc2-f360-4383-8980-6dd31719bb49.png">

I'm not sure it's a good idea to add another `logger` method when there is already a `log`, which could be confusing for the user.

Reply:
Or is that what you meant?

<img width="1153" alt="image" src="https://user-images.githubusercontent.com/8198408/196106625-9916e352-4de8-4fc6-aa7f-f825c9b2d568.png">

Reply:
> Or is that what you meant?

Yes

Reply:
Done